cumulus/test/service/src/lib.rs

```diff
@@ -0,0 +1,627 @@
+use sp_blockchain::HeaderBackend;
+use sp_core::{Pair, H256};
+use sp_keyring::Sr25519Keyring;
+use sp_runtime::{codec::Encode, generic, traits::BlakeTwo256};
+use sp_trie::PrefixedMemoryDB;
+use substrate_test_client::{
+	BlockchainEventsExt, RpcHandlersExt, RpcTransactionError, RpcTransactionOutput,
+};
+
+use cirrus_client_service::{prepare_node_config, start_executor, StartExecutorParams};
+use cirrus_node_primitives::CollatorPair;
+use cirrus_test_runtime::{opaque::Block, Hash, RuntimeApi};
+use cumulus_client_consensus_relay_chain::PrimaryChainConsensus;
+
+pub use cirrus_test_runtime as runtime;
+pub use sp_keyring::Sr25519Keyring as Keyring;
+
+/// The signature of the announce block fn.
+pub type AnnounceBlockFn = Arc<dyn Fn(Hash, Option<Vec<u8>>) + Send + Sync>;
+
+/// Native executor instance.
+pub struct RuntimeExecutor;
+
+impl sc_executor::NativeExecutionDispatch for RuntimeExecutor {
+	type ExtendHostFunctions = ();
+
+	fn dispatch(method: &str, data: &[u8]) -> Option<Vec<u8>> {
+		cirrus_test_runtime::api::dispatch(method, data)
+	}
+
+	fn native_version() -> sc_executor::NativeVersion {
+		cirrus_test_runtime::native_version()
+	}
+}
+
+/// The client type being used by the test service.
+pub type Client =
+	TFullClient<Block, runtime::RuntimeApi, sc_executor::NativeElseWasmExecutor<RuntimeExecutor>>;
+
+/// Starts a `ServiceBuilder` for a full service.
+///
+/// Use this macro if you don't actually need the full service, but just the builder in order to
+/// be able to perform chain operations.
+pub fn new_partial(
+	config: &mut Configuration,
+) -> Result<
+	PartialComponents<
+		Client,
+		TFullBackend<Block>,
+		(),
+		sc_consensus::import_queue::BasicQueue<Block, PrefixedMemoryDB<BlakeTwo256>>,
+		sc_transaction_pool::FullPool<Block, Client>,
+		(),
+	>,
+	sc_service::Error,
+> {
+	let executor = sc_executor::NativeElseWasmExecutor::<RuntimeExecutor>::new(
+		config.wasm_method,
+		config.default_heap_pages,
+		config.max_runtime_instances,
+		config.runtime_cache_size,
+	);
+
+	let (client, backend, keystore_container, task_manager) =
+		sc_service::new_full_parts::<Block, RuntimeApi, _>(config, None, executor)?;
+	let client = Arc::new(client);
+
+	let registry = config.prometheus_registry();
+
+	let transaction_pool = sc_transaction_pool::BasicPool::new_full(
+		config.transaction_pool.clone(),
+		config.role.is_authority().into(),
+		config.prometheus_registry(),
+		task_manager.spawn_essential_handle(),
+		client.clone(),
+	);
+
+	let import_queue = cumulus_client_consensus_relay_chain::import_queue(
+		client.clone(),
+		client.clone(),
+		|_relay_parent, _validation_data| async {
+			let time = sp_timestamp::InherentDataProvider::from_system_time();
+			Ok(time)
+		},
+		&task_manager.spawn_essential_handle(),
+		registry,
+	)?;
+
+	let params = PartialComponents {
+		backend,
+		client,
+		import_queue,
+		keystore_container,
+		task_manager,
+		transaction_pool,
+		select_chain: (),
+		other: (),
```

Comment:
Do we even need these if they are unused?

Reply:
They are the fields of `sc_service::PartialComponents`, which can not be omitted if I understand you correctly.